apps/editing-toolkit/editing-toolkit-plugin/starter-page-templates/page-template-modal/components/sidebar-modal-opener.js

```diff
@@ -13,26 +11,28 @@ import { PageTemplatesPlugin } from '../index';
 class SidebarModalOpener extends Component {
 	state = {
 		isWarningOpen: false,
+		isOpenFromSidebar: false,
 	};
 
 	toggleTemplateModal = () => {
-		this.props.setIsOpen( ! this.props.isOpen );
+		this.setState( { isOpenFromSidebar: ! this.state.isOpenFromSidebar } );
```

Comment:
If the new state depends on the current state then it's best to use the callback version of `setState`

```suggestion
		this.setState( state => ( { isOpenFromSidebar: ! state.isOpenFromSidebar } ) );
```

Reply:
Good call! 👍